react/tic-tac-toe/src/redux/game/actions.js

```diff
@@ -1,32 +1,31 @@
+import { completeTypes, createTypes, withPostFailure } from 'redux-recompose';
+import { SubmissionError } from 'redux-form';
+
 import matchesService from '~services/MatchesService';
 
 import localStorageService from '~services/LocalStorageService';
 
-import { SubmissionError } from 'redux-form';
-
 import ERRORS from '~constants/errors';
 
-export const actions = {
-  MATCHES: '@@GAME/MATCHES',
-  MATCHES_SUCCESS: '@@GAME/MATCHES_SUCCESS',
-  MATCHES_FAILURE: '@@GAME/MATCHES_FAILURE',
-  GAME_SETTIGNS: '@@GAME/GAME_SETTIGNS'
-};
+import { GAME_SETTINGS, MATCHES_TARGET, SETTINGS_TARGET } from './constants';
+
+export const actions = createTypes(completeTypes(['MATCHES'], ['GAME_SETTINGS']), '@@GAME');
 
 const actionsCreators = {
-  getMatches: () => async dispatch => {
-    dispatch({ type: actions.MATCHES });
-    const response = await matchesService.getMatches();
-    if (response.ok) {
-      dispatch({ type: actions.MATCHES_SUCCESS, payload: response.data });
-    } else {
-      dispatch({ type: actions.MATCHES_FAILURE, payload: ERRORS[response.problem] });
-      throw new SubmissionError({ _error: ERRORS[response.problem] });
-    }
-  },
+  getMatches: () => ({
+    type: actions.MATCHES,
+    target: MATCHES_TARGET,
+    service: matchesService.getMatches,
+    failureSelector: response => ERRORS[response.problem],
+    injections: [
+      withPostFailure((dispatch, response) => {
+        throw new SubmissionError({ _error: ERRORS[response.problem] });
+      })
+    ]
+  }),
   setGameSettings: value => dispatch => {
-    localStorageService.setValue(actions.GAME_SETTIGNS, { ...value });
-    dispatch({ type: actions.GAME_SETTIGNS, payload: value });
+    localStorageService.setValue(GAME_SETTINGS, { ...value });
+    dispatch({ type: actions.GAME_SETTINGS, target: SETTINGS_TARGET, payload: value });
```

Comment:
I think I commented in a previous PR, but just in case: I suggest you name the action something clearer. Maybe `SET_SETTINGS`?